File: 231. Power of Two/Approach04.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    bool isPowerOfTwo(int n) {
        if (n <= 0) {
            return false;
        }
        int cnt = 0;
        while(n){
            if(n&1){
                cnt++;
            }
            n = n>>1;
            if(cnt==2){return false;}
        }
        return true;
    }
};

File: 231. Power of Two/Approach04.py
from typing import *

class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if(n<=0):
            return False
        cnt = 0
        while(n):
            if(n&1):
                cnt += 1
            n = n>>1
            if(cnt==2):
                return False
        return True
        